backend/sirius-web-compatibility/src/main/java/org/eclipse/sirius/web/compat/diagrams/ImageNodeSidesConverter.java

```diff
@@ -0,0 +1,29 @@
+package org.eclipse.sirius.web.compat.diagrams;
+
+import com.google.common.collect.Lists;
+
+import java.util.List;
+
+import org.eclipse.sirius.diagram.description.style.Side;
+import org.eclipse.sirius.web.diagrams.ImageNodeSide;
+
+/**
+ * Used to convert Sirius image node forbidden sides
+ * to Sirius Web image node authorized sides.
+ *
+ * @author Koi-tsk
+ */
+public class ImageNodeSidesConverter {
+
+    public List<ImageNodeSide> getAuthorizedSides(List<Side> forbiddenSides) {
+        List<ImageNodeSide> portSides = Lists.newArrayList(ImageNodeSide.NORTH, ImageNodeSide.EAST, ImageNodeSide.SOUTH, ImageNodeSide.WEST);
+        if (forbiddenSides.isEmpty()) {
+            return List.of(ImageNodeSide.UNDEFINED);
```

Comment:
Does `UNDEFINED` need to exist? Why not just an empty list?

Reply:
I used **UNDEFINED** according to the format of elk，and the effect is the same as an empty list.

Reply:
So `UNDEFINED` is not useful on `ImageNodeSide` and it can be removed? I have not yet fully understood why the API of `sirius-web-diagrams` needs to contain that piece of information. From what I see, the authorized sides on which a border node can appear is only a metadata. This information should be retrievable from the `NodeDescription`, I suspect that this information appears on the `NodeImageStyle` because it's a convenient POJO to store and retrieve it later and not because it's relevant.